src/decode/mod.rs

```diff
@@ -9,9 +9,15 @@ pub(crate) mod libipt;
 #[cfg(decoder_libipt)]
 use libipt::LibIPTTraceDecoder;
 
+#[cfg(decoder_ykpt)]
+mod ykpt;
+#[cfg(decoder_ykpt)]
+use ykpt::YkPTTraceDecoder;
+
 #[derive(Clone, Copy, Debug, EnumIter)]
 pub enum TraceDecoderKind {
     LibIPT,
+    YkPT,
```

Comment:
Shouldn't this be `#[cfg(decoder_ykpt)]` too?

Reply:
No, the user should still be able to request any given decoder, even if it isn't available.

Reply:
What does it mean to request a decoder that isn't available?

Reply:
The decoders are conditionally compiled in, but we still allow the user of the API to request any of them when instantiating a decoder.

So if the user does:
```
TraceDecoderBuilder::new().kind(TraceDecoderKind::LibIPT).build();
```

And `hwtracer` was built without libipt, then this would return `Err(HWTracerError::DecoderNotAvailable)`.

If we were to make (e.g.) `TraceDecoderKind::LibIPT` conditionally compiled in too, then that would force consumers of our API to also have compile time switches for hwtracer's decoders and encoders in their build systems too, thus spreading compile time knobs up the dependency chain. This seems undesirable to me.

Reply:
It's not important now but FWIW, I think trading compile-time errors for run-time errors is suboptimal: I would rather know at compile-time that my code won't do what I hope than wait until runtime.

Reply:
That's fine as long as you don't mind imposing conditional compilation on the consumers of the library.

If `DecoderKind::LibIPT` isn't compiled in, then that constant can't be used in consumers, and they too will have to conditionally compile it's use. I think that's a poor API personally.

If you still feel strongly, we can change it later.

Reply:
> That's fine as long as you don't mind imposing conditional compilation on the consumers of the library.

I not only don't mind, I actively *want* to do this!

But, yes, a later PR is fine for this.